controllers/user_controller.go

```diff
@@ -0,0 +1,428 @@
+		err = r.S3Client.DeleteUser(userResource.Spec.Name)
+		if err != nil {
+
+		}
+		return handleS3UserCreation(ctx, userResource, r)
+	} else if err != nil {
+		logger.Error(err, "Could not locate secret", "secret", userResource.Spec.Name)
+		return r.setS3UserStatusConditionAndUpdate(ctx, userResource, "OperatorFailed", metav1.ConditionFalse, "SecretCreationFailed",
+			fmt.Sprintf("Creation of k8s secrets [%s] has failed", userResource.Spec.Name), err)
+	}
+
+	secretKeyValid, err := r.S3Client.CheckUserCredentialsValid(userResource.Spec.Name, userResource.Spec.Name, string(secret.Data["secretKey"]))
+	if err != nil {
+		logger.Error(err, "Something went wrong while checking user credential")
+	}
+
+	if !secretKeyValid {
+		logger.Info("Secret for user is invalid")
+		err = r.S3Client.DeleteUser(userResource.Spec.Name)
+		if err != nil {
+
+		}
+		return handleS3UserCreation(ctx, userResource, r)
+	}
+
+	logger.Info("Check user policies is correct")
+	userPolicies, err := r.S3Client.GetUserPolicies(userResource.Spec.Name)
+	policyToDelete := []string{}
+	policyToAdd := []string{}
+	for _, policy := range userPolicies {
+		policyFound := slices.Contains(userResource.Spec.Policies, policy)
+		if !policyFound {
+			logger.Info(fmt.Sprintf("S3User policy definition doesn't contains policy %s", policy))
+			policyToDelete = append(policyToDelete, policy)
+		}
+	}
+
+	for _, policy := range userResource.Spec.Policies {
+		policyFound := slices.Contains(userPolicies, policy)
+		if !policyFound {
+			logger.Info(fmt.Sprintf("S3User policy definition must contains policy %s", policy))
+			policyToAdd = append(policyToAdd, policy)
+		}
+	}
+
+	if len(policyToDelete) > 0 {
+		r.S3Client.RemovePoliciesFromUser(userResource.Spec.Name, policyToDelete)
+	}
+	if len(policyToAdd) > 0 {
+		r.S3Client.AddPoliciesToUser(userResource.Spec.Name, policyToAdd)
+	}
+
+	logger.Info("Check user groups is correct")
+	userGroups, err := r.S3Client.GetUserGroups(userResource.Spec.Name)
+	groupsToDelete := []string{}
+	groupsToAdd := []string{}
+	for _, group := range userGroups {
+		groupFound := slices.Contains(userResource.Spec.Groups, group)
+		if !groupFound {
+			logger.Info(fmt.Sprintf("S3User groups definition must contains group %s", group))
+
+			groupsToDelete = append(groupsToDelete, group)
+		}
+	}
+
+	for _, group := range userResource.Spec.Groups {
+		groupFound := slices.Contains(groupsToAdd, group)
+		if !groupFound {
+			logger.Info(fmt.Sprintf("S3User groups definition must contains group %s", group))
+			groupsToAdd = append(groupsToAdd, group)
+		}
+	}
+
+	if len(groupsToDelete) > 0 {
+		r.S3Client.RemoveGroupsFromUser(userResource.Spec.Name, groupsToDelete)
+	}
+	if len(groupsToAdd) > 0 {
+		r.S3Client.AddGroupsToUser(userResource.Spec.Name, groupsToAdd)
+	}
+
+	logger.Info("User was reconcile without error")
+
+	return r.setS3UserStatusConditionAndUpdate(ctx, userResource, "OperatorSucceeded", metav1.ConditionTrue, "UserUpdated",
```

Comment:
Maybe replace `OperatorSucceeded` with `Reconciled`?
The condition is about the resource, not about the operator.